site/_plugins/algolia_hooks.rb

```diff
@@ -8,12 +8,32 @@ def self.should_be_excluded?(filepath)
       end
 
       def self.before_indexing_each(record, _node, _context)
+        if _node != nil
+
+          # Simplify h1, h2, h3, h4, h5 and h6 indexing in case of embedded elements.
+          # - Specifically, the h1 in release notes includes an inner <a> that does not need to be indexed.
+          if %w[h1 h2 h3 h4 h5 h6].include? _node.name
+            record[:html] = _node.text.strip
+            record[:text] = _node.text
+          end
+
+          # Collapse JSON for better searching.
+          # - Sort of a variation of https://github.com/algolia/jekyll-algolia/issues/63#issuecomment-380863265
+          #   .content, .text, .to_str, and .inner_text are all aliases.
+          # - Do we have to also modify record[:html] ?
+          if _node.name == 'code' && _node.attr('data-lang') == 'json'
+            record[:html] = _node.text.strip
```

Comment:
You don't have to do anything with the `html` value, we delete them all on line 40.

Reply:
Good point. Maybe I need to go read how the record affects the index so I better understand this part.